src/platform/web.rs

```diff
@@ -436,3 +503,192 @@ impl Display for CustomCursorError {
+    fn request_lock(&self, orientation: OrientationLock) -> OrientationLockFuture;
+
+    /// Unlock the screen orientation.
+    ///
+    /// Will fail if a locking call is in progress.
+    fn unlock(&self) -> Result<(), OrientationLockError>;
+
+    /// Returns whether this [`MonitorHandle`] was created using detailed monitor permissions.
+    ///
+    /// See [`ActiveEventLoop::request_detailed_monitor_permission()`].
+    fn is_detailed(&self) -> bool;
+}
+
+impl MonitorHandleExtWeb for MonitorHandle {
+    fn is_internal(&self) -> Option<bool> {
+        self.inner.is_internal()
+    }
+
+    fn orientation(&self) -> OrientationData {
+        self.inner.orientation()
+    }
+
+    fn request_lock(&self, orientation_lock: OrientationLock) -> OrientationLockFuture {
+        OrientationLockFuture(self.inner.request_lock(orientation_lock))
+    }
+
+    fn unlock(&self) -> Result<(), OrientationLockError> {
+        self.inner.unlock()
+    }
+
+    fn is_detailed(&self) -> bool {
+        self.inner.is_detailed()
+    }
+}
+
+/// Screen orientation data.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub struct OrientationData {
+    /// The orientation.
+    pub orientation: Orientation,
+    /// [`true`] if the [`orientation`](Self::orientation) is flipped upside down.
+    pub flipped: bool,
+    /// [`true`] if the [`Orientation`] is the most natural one for the screen regardless of being
+    /// flipped. Computer monitors are commonly naturally landscape mode, while mobile phones
+    /// are commonly naturally portrait mode.
+    pub natural: bool,
+}
+
+/// Screen orientation.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum Orientation {
+    /// The screen's aspect ratio has a width greater than the height.
+    Landscape,
+    /// The screen's aspect ratio has a height greater than the width.
+    Portrait,
+}
+
+/// Screen orientation lock options. Reoresents which orientations a user can use.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+pub enum OrientationLock {
+    /// User is free to use any orientation.
+    Any,
+    /// User is locked to the most upright natural orientation for the screen. Computer monitors
+    /// are commonly naturally landscape mode, while mobile phones are commonly
+    /// naturally portrait mode.
+    Natural,
+    /// User is locked to landscape mode.
+    Landscape {
+        /// - [`None`]: User is locked to both upright or upside down landscape mode.
+        /// - [`false`]: User is locked to upright landscape mode.
+        /// - [`false`]: User is locked to upside down landscape mode.
+        flipped: Option<bool>,
+    },
+    /// User is locked to portrait mode.
+    Portrait {
+        /// - [`None`]: User is locked to both upright or upside down portrait mode.
+        /// - [`false`]: User is locked to upright portrait mode.
+        /// - [`false`]: User is locked to upside down portrait mode.
+        flipped: Option<bool>,
+    },
```

Comment:
Would use `Orientation` with all the flipped variants added as extension to this.

Reply:
This wouldn't work, because it wouldn't cover `Portrait { flipped: None }` for example.